presto-docs/src/main/sphinx/spill/overview.rst

```diff
@@ -0,0 +1,84 @@
+.. contents::
+    :local:
+    :backlinks: none
+    :depth: 1
+
+Overview
+--------
+
+In the case of memory intensive operations, Presto allows offloading
+intermediate operator results to disk. The goal of this mechanism is to
+enable execution of queries that require amounts of memory exceeding per query
+or per node limits.
+
+The mechanism is similar to OS level page swapping. However, it is
+implemented on the application level to address specific needs of Presto.
+
+Properties related to spilling are described in :ref:`tuning-spilling`.
+
+
+Memory management and spill
+---------------------------
+
+By default, Presto manages memory needed by query runners manually to guarantee
+fairness in allocating memory to queries and prevent saturation of memory by a
+single query (could result in failure of other queries). In the basic case,
+if the memory requested by the query execution exceeds session properties
+``query_max_memory`` or ``query_max_memory_per_node``, the query is killed. This
```

Comment:
This parameters are described in Teradata's documentation: https://github.com/Teradata/presto/blob/teradata-docs/presto-docs/src/main/sphinx/admin/properties.rst#querymax-memory